src/redux/runtime/runtime-slice.ts

```diff
@@ -70,19 +71,21 @@ const runtimeSlice = createSlice({
     name: 'runtime',
     initialState,
     reducers: {
-        setSelected: (state, action: PayloadAction<string[]>) => {
-            state.selected = action.payload;
+        setSelected: (state, action: PayloadAction<Set<Id>>) => {
+            state.selected = new Set<Id>(action.payload);
```

Comment:
My fault. This should be just `state.selected = action.payload`. Better test this to see if we can set `selected` without `structuredClone` or `new Set()`;